files/t/qsub-commands.t

```diff
@@ -121,15 +124,24 @@ is(join(" ", @$newcommand), $txt, "expected command after parse_script without e
 # insert shebang
 {
     local $ENV{SHELL} = '/some/shell';
-    my $stdin = "#\n#PBS -l abd -o stdout.\${PBS_JOBID}..\$PBS_JOBID\n#\n#PBS -e /abc -N def\ncmd\n";
+    my $stdin = "#\n#PBS -l nodes=123 -o stdout.\${PBS_JOBID}..\$PBS_JOBID\n#\n#PBS -e /abc -N def\ncmd\n";
+    diag "replace PBS_JOBID command ", explain($command), " defaults ", explain($defaults), " stdin '$stdin'";
     ($newtxt, $newcommand) = parse_script($stdin, $command, $defaults);
     is(join(" ", @$newcommand),
        "$sbatch --nodes=2 --ntasks=8 --ntasks-per-node=4 --chdir=$ENV{HOME} --export=NONE --get-user-env=60L",
        "expected command after parse_script with eo");
-    is($newtxt, "#!/some/shell\n#\n#PBS -l abd -o ".getcwd."/stdout.%A..%A\n#\n#PBS -e /abc -N def\ncmd\n",
+    is($newtxt, "#!/some/shell\n#\n#SBATCH --nodes=123\n#PBS -o ".getcwd."/stdout.%A..%A\n#\n#PBS -e /abc -N def\ncmd\n",
        "PBS_JOBID replaced");
 }
 
+my $stdin = "#!/bin/something\n#PBS -l nodes=123:ppn=456:gpus=1 -o stdout -l mem=112233\n#\n#PBS -l gpus=2 -l mem=223344 -N def\ncmd\n";
```

Comment:
Maybe also test with something like `nodes=3:ppn=5:gpus=3` (should result in `--gres=gpu:3`, not `gpu:9`)?

Reply:
And `nodes=X:ppn=Y,gpus=Z` is also valid input, so test with that too?

Reply:
no real need to test that, it's already tested as part of the commandline parsing